extensions/VisualEditor/wikia/ApiAddMediaPermanent.php

```diff
@@ -41,7 +43,9 @@ private function executeImage() {
 			$file = new LocalFile( $title, RepoGroup::singleton()->getLocalRepo() );
 			$file->upload( $tempFile->getPath(), '', $pageText ? $pageText : '' );
 			return array(
-				'title' => $file->getTitle()->getText()
+				'title' => $file->getTitle()->getText(),
```

Comment:
Given that response becomes little but more complex - I suggest that there is only one place which constructs it - because on file object. Something like:
```
if ( $duplicate ) {
    $file = $duplicate;
} else {
    $file = someOtherWayOfGettingFile();
}
return array(
    'title' => $file->getTitle()->getText(),
    'url' => $file->getUrl(),
    'image_id' => $file->getTitle()->getArticleID()
)

```